gateway/api/views.py

```diff
@@ -453,6 +453,18 @@ def retrieve(self, request, pk=None):  # pylint: disable=unused-argument
         tracer = trace.get_tracer("gateway.tracer")
         ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
         with tracer.start_as_current_span("gateway.job.retrieve", context=ctx):
+            job = Job.objects.filter(pk=pk).first()
```

Comment:
If I'm understanding this right, if there was no job before we'd just return the serializer (since we didn't have a check), but now we'd be returning a 404... is that ok? (I don't know this area well enough to say, so figured I'd ask the question)

Reply:
It has been returning 404 when the job doesn't exist or the user is not authorized to access to the job.  Here is the test for it https://github.com/Qiskit/qiskit-serverless/blob/2dbdaf52389edb6b8d54b986f30be8fae844fb06/gateway/tests/api/test_job.py#L94